pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Consumer.java

```diff
@@ -435,6 +437,9 @@ public Future<Void> sendMessages(final List<? extends Entry> entries,
         writeAndFlushPromise.addListener(status -> {
             // only increment counters after the messages have been successfully written to the TCP/IP connection
             if (status.isSuccess()) {
+                if (firstConsumedTimestamp == 0) {
+                    firstConsumedTimestamp =  System.currentTimeMillis();
+                }
```

Comment:
just wondering if this makes sense since `flowPermits` method would get called before messages get sent to a consumer?

Reply:
oh sorry, there are 2 separate fields. Just wondering where `lastConsumedTimestamp` is set. wouldn't it be set in a similar way as `firstConsumedTimestamp`?

Reply:
> oh sorry, there are 2 separate fields. Just wondering where lastConsumedTimestamp is set. wouldn't it be set in a similar way as firstConsumedTimestamp?

IMO, we should also move `lastConsumedTimestamp ` to success sent. :/

Reply:
let me rename it to `firstMessagesSentTimestamp`.